--- a/arch/arm64/boot/dts/amlogic/Makefile
+++ b/arch/arm64/boot/dts/amlogic/Makefile
@@ -30,6 +30,7 @@ dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905x-p212.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905d-p230.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905d-p231.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905d-phicomm-n1.dtb
+dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905d-phicomm-n1-thresh.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905d-sml5442tw.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s805x-p241.dtb
 dtb-$(CONFIG_ARCH_MESON) += meson-gxl-s905w-p281.dtb
new file mode 100644
--- /dev/null
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-phicomm-n1-thresh.dts
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright (c) 2018 He Yangxuan
+ * Copyright (c) 2020 Flippy
+ */
+
+/dts-v1/;
+
+#include "meson-gxl-s905d-phicomm-n1.dts"
+
+&ethmac {
+	snps,force_thresh_dma_mode;
+};
--- a/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-phicomm-n1.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxl-s905d-phicomm-n1.dts
@@ -1,6 +1,8 @@
 // SPDX-License-Identifier: (GPL-2.0+ OR MIT)
 /*
  * Copyright (c) 2018 He Yangxuan
+ * Copyright (c) 2020 Flippy
+ * Copyright (c) 2020 Wgsun0592
  */
 
 /dts-v1/;
@@ -11,6 +13,19 @@ / {
 	compatible = "phicomm,n1", "amlogic,s905d", "amlogic,meson-gxl";
 	model = "Phicomm N1";
 
+	dc_in: regulator-vcc_12v {
+		compatible = "regulator-fixed";
+		regulator-name = "VCC_12V";
+		regulator-min-microvolt = <12000000>;
+		regulator-max-microvolt = <12000000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+
+	aliases {
+		serial1 = &uart_A;
+	};
+
 	cvbs-connector {
 		status = "disabled";
 	};
@@ -33,3 +48,40 @@ &cvbs_vdac_port {
 &usb {
 	dr_mode = "host";
 };
+
+&sd_emmc_a {
+	max-frequency = <100000000>;
+};
+
+&sd_emmc_b {
+	status = "okay";
+};
+
+&external_phy {
+	reset-assert-us =    <30000>;
+	reset-deassert-us = <100000>;
+	reset-gpios = <&gpio GPIOZ_14 GPIO_ACTIVE_LOW>;
+	interrupts = <25 IRQ_TYPE_LEVEL_LOW>;
+};
+
+&ethmac {
+	snps,rxpbl = <0x4>;
+	snps,txpbl = <0x8>;
+	snps,aal;
+};
+
+/* This is connected to the Bluetooth module: */
+&uart_A {
+	status = "okay";
+	pinctrl-0 = <&uart_a_pins>, <&uart_a_cts_rts_pins>;
+	pinctrl-names = "default";
+	uart-has-rtscts;
+
+	bluetooth {
+		compatible = "brcm,bcm43438-bt";
+		shutdown-gpios = <&gpio GPIOX_17 GPIO_ACTIVE_HIGH>;
+		max-speed = <2000000>;
+		clocks = <&wifi32k>;
+		clock-names = "lpo";
+	};
+};
